qa/tasks/cephadm.py

```diff
@@ -1264,13 +1264,17 @@ def registries_add_mirror_to_docker_io(conf, mirror):
         }
     else:
         v2 = config  # type: ignore
-    dockers = [r for r in v2['registry'] if r['prefix'] == 'docker.io']
+    dockers = [
+        r for r in v2['registry'] if
+           r.get('prefix') == 'docker.io' or r.get('location') == 'docker.io'
+    ]
     if dockers:
         docker = dockers[0]
-        docker['mirror'] = [{
-            "location": mirror,
-            "insecure": True,
-        }]
+        if 'mirror' not in docker:
+            docker['mirror'] = [{
+                "location": mirror,
+                "insecure": True,
+            }]
     return v2
```

Comment:
let's see. I'd love to drop this code entirely from the cephadm.py task, if possible and only rely on https://github.com/ceph/ceph-cm-ansible.

Reply:
I'll run a test with that code removed and see how it goes.

Reply:
http://pulpito.front.sepia.ceph.com/dmick-2020-12-19_00:34:42-rados:cephadm:smoke-wip-cephadm-no-mirror-setup-distro-basic-gibba/ with this:

<pre>
diff --git a/qa/tasks/cephadm.py b/qa/tasks/cephadm.py
index 0ff2acf925a..78c1218711a 100644
--- a/qa/tasks/cephadm.py
+++ b/qa/tasks/cephadm.py
@@ -308,8 +308,8 @@ def ceph_bootstrap(ctx, config, registry):

     ctx.cluster.sh('sudo mkdir -p /etc/ceph')
     ctx.cluster.sh('sudo chmod 777 /etc/ceph')
-    if registry:
-        add_mirror_to_cluster(ctx, registry)
+    #if registry:
+    #    add_mirror_to_cluster(ctx, registry)
     try:
         # write seed config
         log.info('Writing seed config...')
</pre>

seems to have worked okay.  We can just purge all that code I suppose.

Reply:
let's make this a new PR then